src/internal/Observable.ts

```diff
@@ -506,15 +488,7 @@ export class Observable<T> implements Subscribable<T> {
  * @param promiseCtor The optional promise constructor to passed by consuming code
  */
 function getPromiseCtor(promiseCtor: PromiseConstructorLike | undefined) {
-  if (!promiseCtor) {
-    promiseCtor = config.Promise || Promise;
-  }
-
-  if (!promiseCtor) {
-    throw new Error('no Promise impl found');
-  }
-
-  return promiseCtor;
+  return promiseCtor ?? config.Promise ?? Promise;
```

Comment:
This was changed to make it lighter weight.

The old code served no purpose that wouldn't be served by `ReferenceError: Promise is not defined`.